node/chainSimulator/chainSimulator.go

```diff
@@ -144,6 +145,11 @@ func setSimulatorRunTypeArguments(args *ArgsChainSimulator) {
 	if args.ChainProcessorFactory == nil {
 		args.ChainProcessorFactory = NewChainHandlerFactory()
 	}
+	if args.GenerateGenesisFile == nil {
```

Comment:
Not necessarily for now, but we definetely need to put all of these func pointers inside a struct holder...

Reply:
Yep, that's for another time...